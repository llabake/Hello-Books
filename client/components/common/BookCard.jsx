import React from 'react';
import { Link } from 'react-router-dom';
import { bookDefaultImage } from '../../helpers/utils';

import leanstart from '../../media/lean start.jpg';

const BookCard = (props) => {
  const { book } = props;
  return (
    <div className="book col s12 m3 l3">
      <div className="card" style={{ minHeight: '250px', maxHeight: '250px' }}>
        <div className="card-image waves-effect waves-block waves-light" style={{ maxHeight: '175px', minHeight: '170px', paddingTop: '10px' }}>
          <Link to={'/book/' + book.id}>
            <img src={book.image || bookDefaultImage}
              style={{ maxHeight: '170px', minHeight: '165px' }}
              alt="Book Image"
              title={book.title} 
            />
          </Link>
        </div>
        <div className="card-content"
          style={{ position: 'absolute', bottom: 0, width: '100%' }}>
          <p><Link to={'/book/' + book.id} className="btn">Details</Link></p>
        </div>
      </div>
      <div className="book-info center-align">
        <div className="book-title"><span>{book.title} </span></div>
        <div className="book-author">by <span>{book.author}</span></div>
      </div>
    </div>

  );
}

export default BookCard;