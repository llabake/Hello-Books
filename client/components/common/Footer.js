import React from 'react';
import { Link } from 'react-router-dom';

const Footer = () => {
  const presentYear = new Date().getFullYear();
  return (
    <footer className="page-footer center">
      <div className="container">
        <div className="row">
          <div className="col s12 m4 l3">
            <h5 className="white-text">My Account</h5>
            <ul className="navigation">
              <li><Link to="/" className="grey-text text-lighten-4">Home</Link></li>
              <li><Link to="/profile" className="white-text text-lighten-4">Profile</Link></li>
              <li><Link to="/favorite" className="grey-text text-lighten-4">Favorite</Link></li>
            </ul>
          </div>
          <div className="col s12 m4 l6">
            <h5 className="white-text">Hello Books</h5>
            <p className="white-text text-lighten-4">Stay Connected. Join our over 5 million users today</p>
          </div>
          <div className="col s12 m4 l3" >
            <h5 className="white-text">Be Social</h5>
            <ul className="list-inline">
              <li>
                <a href="https://facebook.com/hellobooksapp" className="social-btn" ><i className="fa fa-facebook-square fa-2x icon-style" /></a>
              </li>
              <li><a href="https://instagram.com/hellobooksapp" className="social-btn"><i className="fa fa-instagram fa-2x icon-style" /></a></li>
              <li><a href="https://twitter.com/hellobooksapp" className="social-btn" ><i className="fa fa-twitter-square fa-2x icon-style" /></a></li>
              <li><a href="https://pinterest.com/hellobooksapp" className="social-btn"><i className="fa fa-pinterest-square fa-2x icon-style" /></a></li>
            </ul>
          </div>
        </div>
      </div>
      <div className="footer-copyright center-align">
        <div className="container">
          Â© {presentYear} Happy Girl World
      </div>
      </div>
    </footer>
  );
};

export default Footer;

