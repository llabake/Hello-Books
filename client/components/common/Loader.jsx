import React from 'react';

import ajaxLoader from '../../media/ajax-loader.gif';

const Loader = () => {
  return (
    <div
      className="center-align"
      style={{ marginTop: '1em', marginBottom: '1em' }}
    >
      <img src={ajaxLoader} alt="Loading..." />
    </div>
  )
}

export default Loader;