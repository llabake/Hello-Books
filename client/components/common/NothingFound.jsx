import React from 'react';
import PropTypes from 'prop-types';

const NothingFound = ({ text, id }) => {
  return (
    <div
      className="card-panel row center-align"
      id={id}>
      <h4>
        {text}
      </h4>
    </div>
  )
};

NothingFound.propTypes = {
  id: PropTypes.string,
  text: PropTypes.string
};

export default NothingFound;