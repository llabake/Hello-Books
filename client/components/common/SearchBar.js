import React, { Component } from 'react';
import { connect } from 'react-redux';

/**
 * 
 * 
 * @class SearchBar
 * @extends {Component}
 */
class SearchBar extends Component {

  /**
   * Creates an instance of SearchBar.
   * @param {any} props 
   * @memberof SearchBar
   */
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: '',
    };
    
    this.handleChange = this.handleChange.bind(this)
    this.handleSearch = this.handleSearch.bind(this)
    this.redirectToSearch = this.redirectToSearch.bind(this)
  }

  /**
   * @return {object} state object
   * 
   * @param {any} event 
   * @memberof SearchBar
   */
  handleChange (event) {
    this.setState({
      [event.target.name]: event.target.value
    })
  }

  
  /**
   * @returns {array} an array of books that match the search criteria
   * 
   * @param {any} event 
   * @memberof SearchBar
   */
  handleSearch(event) {
    event.preventDefault();
    this.redirectToSearch();
    
  }

  /**
   * @returns {object} redirects to search page
   * 
   * @memberof SearchBar
   */
  redirectToSearch(){
    window.location.href = `/search/${this.state.searchTerm}`
  }


  /**
   * 
   * 
   * @returns {void} 
   * @memberof SearchBar
   */
  render () {
    return (
      <form onSubmit={this.handleSearch}>
        <div className="input-field col m6 s12 l6 " style={{'marginRight': '12px'}}>
          <input
            id="search" 
            type="search"  
            placeholder="Title, author, or ISBN" 
            name='searchTerm' 
            onChange={this.handleChange}
            value={this.state.searchTerm}
          />
          <label className="label-icon icon-sit" htmlFor="search"><a><i className="material-icons">search</i></a></label>
          {/* <label className="label-icon icon-sit" htmlFor="search"><button style={{
            'lineHeight': '15px',
            'letterSpacing': 'undefined',
            'paddingTop': '0px' ,
            'paddingBottom': '0px',
            'color': '#ffffff',
            'paddingLeft': '18px',
            'paddingRight': '18px',
            'borderRadius': '4px',
            'border': '0px',
            'background': '#7c4561',
            'margin': '12px ',
            'marginLeft': '16px' ,
            'height': '48px',
            'textAlign': 'center',
            'display': 'inline-block',
            'float': 'right',
            'whiteSpace': 'nowrap'
            }}>Search</button></label> */}
          <i className="material-icons">close</i>
        </div>
      </form>
    )
  }

}

const mapStateToProps = (state) => {
  return {
    errors: state.errors,
    searchedBooks: state.bookReducer.searchResult
  }
}

export default connect(mapStateToProps, null)(SearchBar);