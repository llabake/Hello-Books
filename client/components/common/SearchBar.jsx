import React, { Component } from 'react';
import { connect } from 'react-redux';

/**
 * 
 * 
 * @class SearchBar
 * @extends {Component}
 */
class SearchBar extends Component {

  /**
   * Creates an instance of SearchBar.
   * @param {any} props 
   * @memberof SearchBar
   */
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: '',
    };

    this.handleChange = this.handleChange.bind(this)
    this.handleSearch = this.handleSearch.bind(this)
    this.redirectToSearch = this.redirectToSearch.bind(this)
  }

  /**
   * @return {object} state object
   * 
   * @param {any} event 
   * @memberof SearchBar
   */
  handleChange(event) {
    this.setState({
      [event.target.name]: event.target.value
    })
  }


  /**
   * @returns {array} an array of books that match the search criteria
   * 
   * @param {any} event 
   * @memberof SearchBar
   */
  handleSearch(event) {
    event.preventDefault();
    this.redirectToSearch();

  }

  /**
   * @returns {object} redirects to search page
   * 
   * @memberof SearchBar
   */
  redirectToSearch() {
    const trimmedSearchTerm = this.state.searchTerm && this.state.searchTerm.trim();
    trimmedSearchTerm ? 
    window.location.href = `/search/${trimmedSearchTerm}`:
    window.location.href = window.location.href 
  }


  /**
   * 
   * 
   * @returns {void} 
   * @memberof SearchBar
   */
  render() {
    return (
      <form onSubmit={this.handleSearch}>
        <div className="input-field col s6 s12 ">
          <input
            // id="search"
            type="search"
            placeholder="Title, Author, Keyword"
            name='searchTerm'
            onChange={this.handleChange}
            value={this.state.searchTerm}
          />
          <label
            className="label-icon icon-sit"
            htmlFor="search">
            <i
              className="material-icons clickable"
              onClick={this.handleSearch}>search
            </i>
          </label>
          <i className="material-icons"
          style={{marginTop: '-6px', fontSize: '1.5rem'}}
          >close</i>
        </div>
      </form>
    )
  }

}

const mapStateToProps = (state) => {
  return {
    errors: state.errors,
    searchedBooks: state.bookReducer.searchResult
  }
}

export default connect(mapStateToProps, null)(SearchBar);