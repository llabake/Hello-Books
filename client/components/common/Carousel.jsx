import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { bookDefaultImage } from '../../helpers/utils';
import { fetchTopFavoriteBooks } from '../../actions/bookAction'
import Loader from '../common/Loader';

/**
 * 
 * 
 * @export
 * @class Carousel
 * @extends {Component}
 */
export class Carousel extends Component {

  /**
   * @returns {array} array of top favorite books
   * @memberof Carousel
   */
  componentWillMount() {
    this.props.fetchTopFavoriteBooks();
  }

  /**
  * 
  * @returns {function} initialises the carousel
  * @memberof Carousel
  */
  componentDidMount() {
    this.initializeCarousel()
  }

  /**
   * @returns {function} initialises the carousel
   * @memberof Carousel
   */
  componentDidUpdate() {
    this.initializeCarousel()
  }

  /**
   * @returns {function} intialise carousel
   * 
   * @memberof Carousel
   */
  initializeCarousel() {
    $('.carousel').carousel({
      dist: 0,
      shift: 0,
      padding: 50,
      indicators: true,
    });
    autoplay()
    /**
     * @returns {function} materialize carousel autoplay
     *
     */
    function autoplay() {
      $('.carousel').carousel('next');
      setTimeout(autoplay, 4500)
    }
  }

  /**
   * 
   * 
   * @returns {Object} html
   * @memberof Carousel
   */
  render() {
    this.initializeCarousel()
    let { topFavoriteBooks, loadingTopFavoritedBooks } = this.props;

    return (
      <div id='my-carousel'>
        {
          loadingTopFavoritedBooks ? <Loader /> : ''
        }
        {!loadingTopFavoritedBooks ? topFavoriteBooks.length ?
          <div className="carousel" style={{ marginTop: '-5em' }}> {
            topFavoriteBooks.map((book, index) => {
              return <Link
                className='carousel-item pointer'
                key={index}
                to={'/book/' + book.id}
                title={book.title}
              ><img
                  src={book.image || bookDefaultImage}
                  style={{ minHeight: '200px', maxHeight: '200px' }}
                />
              </Link>
            })} </div>
          : null : null
        }
      </div>
    )
  }
}

export const mapStateToProps = (state) => {
  return {
    topFavoriteBooks: state.bookReducer.topFavoriteBooks,
    loadingTopFavoritedBooks: state.bookReducer.loadingTopFavoritedBooks
  };
};


export const mapDispatchToProps = (dispatch) => {
  return {
    fetchTopFavoriteBooks: () => dispatch(fetchTopFavoriteBooks()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Carousel);
