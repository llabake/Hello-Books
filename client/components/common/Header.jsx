import React, { Component } from 'react';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux'
import { logout } from '../../actions/userAction'

import SearchBar from './SearchBar';
import SideNav from './SideNav';

/**
 * 
 * 
 * @class Header
 * @param {any} user
 * @extends {Component}
 */
export class Header extends Component {

  /**
   * Creates an instance of Header.
   * @memberof Header
   */
  constructor() {
    super();
    this.handleLogout = this.handleLogout.bind(this);
  }

  /**
 * @returns {object} redirects to home page
 * 
 * @memberof Header
 */
  handleLogout() {
    this.props.logout();
  }

  /**
   * @returns {void}
   * 
   * @memberof Header
   */
  componentDidMount() {
    // Initialize materialize dropdown class
    $('.dropdown-button').dropdown({
      belowOrigin: true,
    });
    $('.button-collapse').sideNav({
      closeOnClick: true,
      draggble: true
    })
    $('.collapsible').collapsible();
  }

  /**
   * @returns {void}
   * 
   * @memberof Header
   */
  componentDidUpdate() {
    // Initialize materialize dropdown class
    $('.dropdown-button').dropdown({
      belowOrigin: true,
    });
  }



  /**
   * 
   * 
   * @param {any} user 
   * @returns {object} admin navigation bar
   * @memberof Header
   */
  renderAdminNavigation(user) {
    if (user.role === "admin") {
      return (
        <li><a className="dropdown-button" id="adminusername" data-activates="dropdown1">{user.username}
          <i className="material-icons large left">account_circle</i>
          <i className="material-icons right">arrow_drop_down</i></a>
          <ul id="dropdown1" className="dropdown-content">
            <li><Link to="/admindashboard">Admin Dashboard</Link></li>
            <li><Link to="/favorite">Favorite Books</Link></li>
            <li><Link to="/profile">Profile</Link></li>
            <li><Link to="/addbook">Add Book</Link></li>
            <li><Link to="/allbooks">All Books</Link></li>
            <li className="divider"></li>
            <li><Link to=""  id='adminLogout' onClick={this.handleLogout}><i className="material-icons ">lock</i>Log Out</Link></li>
          </ul>
        </li>
      )
    }
  }


  /**
   * 
   * @returns {object} index navigation bar
   * 
   * @memberof Header
   */
  renderIndexNavigation() {
    return (
      <li>
        <a className="dropdown-button" data-activates="dropdown1">Hi, User
        <i className="material-icons large left">account_circle</i>
          <i className="material-icons right">arrow_drop_down</i>
        </a>
        <ul id="dropdown1" className="dropdown-content">
          <li><Link to="/signup">Sign Up</Link></li>
          <li><Link to="/signin">Sign In</Link></li>
          <li className="divider"></li>
        </ul>
      </li>
    )

  }

  /**
   * 
   * @param {any} user
   * 
   * @returns {object} authenticated user nav bar
   * 
   * @memberof Header
   */
  renderAuthUserNavigation(user) {
    return (
      <li>
        <a className="dropdown-button" id="username" data-activates="dropdown1">{user.username}
          <i className="material-icons large left">account_circle</i>
          <i className="material-icons right">arrow_drop_down</i>
        </a>

        <ul id="dropdown1" className="dropdown-content">
          <li><Link to="/allbooks">All Books</Link></li>
          <li><Link to="/favorite">Favorite Books</Link></li>
          <li><Link to="/profile">Profile</Link></li>
          <li className="divider"></li>
          <li>
            <NavLink to=""
            id='userLogout'
              onClick={this.handleLogout}>Log Out
          </NavLink>
          </li>
        </ul>
      </li>
    )
  }

  /**
   *
   * @memberof Header
   * 
   * @param {any} user
   * 
   * @returns {Object} nav bar based on user's role
   */
  renderNavigation = (user) => user.role === "admin"
    ? this.renderAdminNavigation(user)
    : this.renderAuthUserNavigation(user)

  /**
   * 
   * @returns {void}
   * 
   * @memberof Header
   */
  render() {
    const { user, authenticated } = this.props
    return (
      <header>
        <div className="navbar-fixed">
          <nav>
            <div className="nav-wrapper">
              <NavLink
                to="/"
                className="brand-logo left adjust">HelloBooks
              </NavLink>
              <a href="#" data-activates="mobile-demo" className="button-collapse">
                <i className="material-icons">menu</i>
              </a>
              <ul id="nav-mobile" className="right hide-on-med-and-down">
                <li>
                  <SearchBar />
                </li>
                {
                  authenticated && user
                    ? this.renderNavigation(user) :
                    this.renderIndexNavigation()
                }
              </ul>
            </div>
          </nav>
        </div>
        <SideNav
          user={user}
          handleLogout={this.handleLogout}
          authenticated={authenticated}
        />
      </header>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    errors: state.errors,
    user: state.userReducer.authUser,
    authenticated: state.userReducer.authenticated,
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    logout: () => dispatch(logout())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
