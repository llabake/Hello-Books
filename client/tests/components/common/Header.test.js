// import React from 'react'
// import { createStore } from 'redux'
// import { Provider } from 'react-redux'
// import { MemoryRouter } from 'react-router-dom'

// import reducer from '../../../reducers'
// import { Header } from '../../../components/common/Header'

// test('The header component mounts correctly if all props it needs are passed into it.', () => {
//   const props = {
//     authenticated: false,
//     user: {},
//     errors: {}
//   }

//   const store = createStore(reducer)
//   const wrapper = mount(
//     <Provider store={store}>
//       <MemoryRouter>
//         <Header {...props} />
//       </MemoryRouter>
//     </Provider>)
// })

import React from 'react'
import { NavLink } from 'react-router-dom'
import { Header, mapStateToProps, mapDispatchToProps} from '../../../components/common/Header'
import userData from '../../mocks/userData'

test('The header component mounts correctly if all props it needs are passed into it.', () => {
  const props = {
    authenticated: false,
    user: {},
    errors: {}
  }

  const wrapper = shallow(<Header {...props} />)
  expect(wrapper).toMatchSnapshot()
})

test('the user details are displayed when the user is logged in', () => {
  const props = {
    authenticated: true,
    user: userData.user,
    errors: {}
  }
  const wrapper = shallow(<Header {...props} />)
  const usernameDropdown = wrapper.find('#username')
  expect(usernameDropdown.text()).toMatch(`${props.user.username}account_circlearrow_drop_down`)
})

test('the admin details are displayed when the user is logged in as an admin', () => {
  const props = {
    authenticated: true,
    user: userData.admin,
    errors: {}
  }
  const wrapper = shallow(<Header {...props} />)
  const usernameDropdown = wrapper.find('#username')
  expect(usernameDropdown.text()).toMatch(`${props.user.username}account_circlearrow_drop_down`)
});

test('calls component did update to reinitialize materialize dropdown ', () => {
  const props = {
    authenticated: false,
    user: {},
    errors: {},
    logout: jest.fn()
  };
  const nextProps = {
    authenticated: true,
    user: userData.user
  };
  const spy = jest.spyOn(Header.prototype, 'componentDidUpdate')
  const wrapper = shallow(<Header {...props} />)
  expect(wrapper).toBeDefined();
  wrapper.setProps(nextProps);
  expect(spy).toHaveBeenCalled();
  expect(wrapper).toMatchSnapshot();
})

test('user can logout when they click on logout from the dropdown', () => {
  const props = {
    authenticated: true,
    user: userData.user,
    errors: {},
    logout: jest.fn()
  };
  const wrapper = shallow(<Header {...props} />);
  const handleLogoutSpy = jest.spyOn(wrapper.instance(), 'handleLogout');
  wrapper.instance().handleLogout();
  expect(handleLogoutSpy).toHaveBeenCalledTimes(1);
});

test('mapDispatchToProps', () => {
  const dispatch = jest.fn();
  expect(mapDispatchToProps(dispatch)).toHaveProperty('logout');

  const { logout } = mapDispatchToProps(dispatch);
  logout();
  expect(dispatch).toHaveBeenCalled();
});