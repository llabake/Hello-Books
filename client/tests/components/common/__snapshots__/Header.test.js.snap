// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The header component mounts correctly if all props it needs are passed into it. 1`] = `
<header>
  <div
    className="navbar-fixed"
  >
    <nav>
      <div
        className="nav-wrapper"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="brand-logo left adjust"
          to="/"
        >
          HelloBooks
        </NavLink>
        <a
          className="button-collapse"
          data-activates="mobile-demo"
          href="#"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
        <ul
          className="right hide-on-med-and-down"
          id="nav-mobile"
        >
          <li>
            <Connect(SearchBar) />
          </li>
          <li>
            <a
              className="dropdown-button"
              data-activates="dropdown1"
            >
              Hi, User
              <i
                className="material-icons large left"
              >
                account_circle
              </i>
              <i
                className="material-icons right"
              >
                arrow_drop_down
              </i>
            </a>
            <ul
              className="dropdown-content"
              id="dropdown1"
            >
              <li>
                <Link
                  replace={false}
                  to="/signup"
                >
                  Sign Up
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/signin"
                >
                  Sign In
                </Link>
              </li>
              <li
                className="divider"
              />
            </ul>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <SideNav
    authenticated={false}
    handleLogout={[Function]}
    user={Object {}}
  />
</header>
`;

exports[`calls component did update to reinitialize materialize dropdown  1`] = `
<header>
  <div
    className="navbar-fixed"
  >
    <nav>
      <div
        className="nav-wrapper"
      >
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="brand-logo left adjust"
          to="/"
        >
          HelloBooks
        </NavLink>
        <a
          className="button-collapse"
          data-activates="mobile-demo"
          href="#"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
        <ul
          className="right hide-on-med-and-down"
          id="nav-mobile"
        >
          <li>
            <Connect(SearchBar) />
          </li>
          <li>
            <a
              className="dropdown-button"
              data-activates="dropdown1"
              id="username"
            >
              labake
              <i
                className="material-icons large left"
              >
                account_circle
              </i>
              <i
                className="material-icons right"
              >
                arrow_drop_down
              </i>
            </a>
            <ul
              className="dropdown-content"
              id="dropdown1"
            >
              <li>
                <Link
                  replace={false}
                  to="/allbooks"
                >
                  All Books
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/favorite"
                >
                  Favorite Books
                </Link>
              </li>
              <li>
                <Link
                  replace={false}
                  to="/profile"
                >
                  Profile
                </Link>
              </li>
              <li
                className="divider"
              />
              <li>
                <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  id="userLogout"
                  onClick={[Function]}
                  to=""
                >
                  Log Out
                </NavLink>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <SideNav
    authenticated={true}
    handleLogout={[Function]}
    user={
      Object {
        "active": true,
        "email": "labake.labs@gmail.com",
        "id": 3,
        "role": "normal",
        "username": "labake",
      }
    }
  />
</header>
`;
